extensions/sql/contract-negotiation-store/src/main/java/org/eclipse/dataspaceconnector/sql/contractnegotiation/store/SqlContractNegotiationStore.java

```diff
@@ -0,0 +1,311 @@
+                toJson(negotiation.getContractOffers()),
+                toJson(negotiation.getTraceContext()));
+
+
+    }
+
+    @Nullable
+    private <T> T single(List<T> list) {
+        if (list.size() > 1) {
+            throw new IllegalStateException(getMultiplicityError(1, list.size()));
+        }
+
+        return list.isEmpty() ? null : list.get(0);
+    }
+
+    private ContractAgreement mapContractAgreement(ResultSet resultSet) throws SQLException {
+        return ContractAgreement.Builder.newInstance()
+                .id(resultSet.getString(statements.getContractAgreementIdColumn()))
+                .providerAgentId(resultSet.getString(statements.getProviderAgentColumn()))
+                .consumerAgentId(resultSet.getString(statements.getConsumerAgentColumn()))
+                .asset(Asset.Builder.newInstance().id(resultSet.getString(statements.getAssetIdColumn())).build())
+                .policy(Policy.Builder.newInstance().id(resultSet.getString(statements.getPolicyIdColumn())).build())
+                .contractStartDate(resultSet.getLong(statements.getStartDateColumn()))
+                .contractEndDate(resultSet.getLong(statements.getEndDateColumn()))
+                .contractSigningDate(resultSet.getLong(statements.getSigningDateColumn()))
+                //todo
+                .build();
+    }
+
+    private String getMultiplicityError(int expectedSize, int actualSize) {
+        return format("Expected to find %d items, but found %d", expectedSize, actualSize);
+    }
+
+    private ContractNegotiation mapContractNegotiation(ResultSet resultSet) throws SQLException, JsonProcessingException {
+        return ContractNegotiation.Builder.newInstance()
+                .id(resultSet.getString(statements.getIdColumn()))
+                .counterPartyId(resultSet.getString(statements.getCounterPartyIdColumn()))
+                .counterPartyAddress(resultSet.getString(statements.getCounterPartyAddressColumn()))
+                .protocol(resultSet.getString(statements.getProtocolColumn()))
+                .correlationId(resultSet.getString(statements.getCorrelationIdColumn()))
+                .contractAgreement(extractContractAgreement(resultSet))
+                .state(resultSet.getInt(statements.getStateColumn()))
+                .stateCount(resultSet.getInt(statements.getStateCountColumn()))
+                .stateTimestamp(resultSet.getLong(statements.getStateTimestampColumn()))
+                .contractOffers(fromJson(resultSet.getString(statements.getContractOffersColumn()), new TypeReference<>() {
+                }))
+                .errorDetail(resultSet.getString(statements.getErrorDetailColumn()))
+                .traceContext(fromJson(resultSet.getString(statements.getTraceContextColumn()), new TypeReference<>() {
+                }))
+                .build();
+    }
+
+    private String toJson(Object object) {
+        try {
+            return objectMapper.writeValueAsString(object);
```

Comment:
`TypeManager` already offer the `writeValueAsString` that handles checked exception, could be preferred to `ObjectMapper`